src/lnp/java/teammates/lnp/cases/InstructorStudentCascadingUpdateLNPTest.java

```diff
@@ -0,0 +1,360 @@
+                    }
+                }
+
+                return feedbackResponseComments;
+            }
+
+            @Override
+            protected Map<String, InstructorAttributes> generateInstructors() {
+                Map<String, InstructorAttributes> instructors = new HashMap<>();
+
+                instructors.put(INSTRUCTOR_ID,
+                        InstructorAttributes.builder(COURSE_ID, INSTRUCTOR_EMAIL)
+                                .withGoogleId(INSTRUCTOR_ID)
+                                .withName(INSTRUCTOR_NAME)
+                                .withRole("Co-owner")
+                                .withIsDisplayedToStudents(true)
+                                .withDisplayedName("Co-owner")
+                                .withPrivileges(new InstructorPrivileges(
+                                        Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER))
+                                .build()
+                );
+
+                return instructors;
+            }
+
+            @Override
+            public List<String> generateCsvHeaders() {
+                List<String> headers = new ArrayList<>();
+
+                headers.add("loginId");
+                headers.add("isAdmin");
+                headers.add("courseId");
+                headers.add("enrollData");
+
+                return headers;
+            }
+
+            @Override
+            public List<List<String>> generateCsvData() {
+                DataBundle dataBundle = loadDataBundle(getJsonDataPath());
+                List<List<String>> csvData = new ArrayList<>();
+
+                dataBundle.instructors.forEach((key, instructor) -> {
+                    List<String> csvRow = new ArrayList<>();
+
+                    csvRow.add(instructor.googleId);
+                    csvRow.add("yes");
+                    csvRow.add(instructor.courseId);
+
+                    // Create and add student enrollment data with a team number corresponding to each section number
+                    List<StudentsEnrollRequest.StudentEnrollRequest> enrollRequests = new ArrayList<>();
+                    int startIndex = csvTestDataIndex * NUM_STUDENTS_PER_SECTION;
+
+                    for (int i = startIndex; i < startIndex + NUM_STUDENTS_PER_SECTION; i++) {
+                        String name = instructor.name + ".Student" + (NUM_STUDENTS - i);
+                        String email = STUDENT_NAME_PREFIX + i + STUDENT_EMAIL_SUBFIX;
+                        String team = String.valueOf((NUM_STUDENTS - i) / NUM_STUDENTS_PER_SECTION);
+                        String section = String.valueOf((NUM_STUDENTS - i) / NUM_STUDENTS_PER_SECTION);
+                        String comment = "no comment";
+
+                        enrollRequests.add(
+                                new StudentsEnrollRequest.StudentEnrollRequest(name, email, team, section, comment)
+                        );
+                    }
```

Comment:
Can you elaborate a bit more about how the students differ from the initial enrollment? We want to maximize the number of data write per enrollment for stress test purpose, so might help if you can point out the parts that would incur the most data writes.

Can give some examples of the backend optimization logic, i.e. `if (isResponseIdChanged || isGiverSectionChanged || isRecipientSectionChanged)` would prevent a response comment that was not changed from inducing a data write, and show that this script indeed would push the enrollment logic to the limit? 

Here we assume that data IO overhead outweighs the latency from the data transfer (e.g. if 10kb data takes 100ms to transfer, 100kb data might take 110ms to transfer), and therefore we do not introduce variance in the size of the data transfer but simply the number of data write that could occur.

Reply:
Every student attribute here is different from the existing ones, except for email, so I believe all responses and response comments would be updated

Reply:
In this case, I believe some comments would help as I do no know which part of the student is altered easily.
I do agree that by changing every attributes including teams and sections, the enrollment will trigger the most writes.